### PR TITLE
Refactor vrg-kubeobject functions

### PR DESCRIPTION
**Pull request by raghavendra-talur**
Refactored the following functions
- kubeObjectsRecoveryStartOrResume
- executeRecoverGroup
- kubeObjectsRecover

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
@BenamarMk This PR is ready for review. I have rebased it on top of #1709 so the e2e failures should be gone.

File: internal/controller/vrg_kubeobjects.go [hunk 1]
```diff
@@ -627,16 +602,60 @@ func (v *VRGInstance) getRecoverOrProtectRequest(
...
 func (v *VRGInstance) kubeObjectsRecoveryStartOrResume(
-	result *ctrl.Result, s3StoreAccessor s3StoreAccessor,
-	sourceVrgNamespaceName, sourceVrgName string,
+	result *ctrl.Result, s3ProfileName string,
 	captureToRecoverFromIdentifier *ramen.KubeObjectsCaptureIdentifier,
-	captureRequests, recoverRequests map[string]kubeobjects.Request,
-	veleroNamespaceName string, labels map[string]string, log logr.Logger,
+	log logr.Logger,
 ) error {
+	veleroNamespaceName := v.veleroNamespaceName()
```
Comment (ELENAGER) [inline internal/controller/vrg_kubeobjects.go:633]: If we are already refactoring - I think we can just use v.veleroNamespaceName() in the exact places we are calling to v.reconciler.kubeObjects.ProtectRequestCreate or v.reconciler.kubeObjects.ProtectRequestDelete or v.reconciler.kubeObjects.ProtectRequestGet. But instead we are passing it as a parameter everywhere.

File: internal/controller/vrg_kubeobjects.go [hunk 2]
```diff
@@ -495,76 +495,54 @@ func (v *VRGInstance) kubeObjectsCaptureStatus(status metav1.ConditionStatus, re
 	}
 }
 
-func (v *VRGInstance) kubeObjectsRecover(result *ctrl.Result,
-	s3StoreProfile ramen.S3StoreProfile, objectStorer ObjectStorer,
-) error {
-	if v.kubeObjectProtectionDisabled("recovery") {
-		return nil
-	}
+func (v *VRGInstance) getVRGFromS3Profile(s3ProfileName string) (*ramen.VolumeReplicationGroup, error) {
```
Comment (BenamarMk) [inline internal/controller/vrg_kubeobjects.go:498]: We have this function: `GetLastKnownVRGPrimaryFromS3`. Can we have a common function?

Comment (raghavendra-talur) [reply internal/controller/vrg_kubeobjects.go:498]: @BenamarMk that is a function in DRPC and this is a function in VRG as a method on VRGInstance, can I take up the refactoring in a later PR?

File: internal/controller/vrg_kubeobjects.go (hunk 1 above)
Comment (raghavendra-talur) [reply internal/controller/vrg_kubeobjects.go]: Done

**Review by ELENAGER — APPROVED** [review]
I  like the changes! Code is much better now.

**Review by asn1809 — APPROVED** [review]

**Pull request merged**